operators/pkg/instautoctrl/termination.go

```diff
@@ -179,7 +180,23 @@ func (r *InstanceTerminationReconciler) TerminateInstance(ctx context.Context, i
 	log := ctrl.LoggerFrom(ctx).WithName("termination")
 	log.Info("terminating instance")
 
-	instance.SetLabels(forge.InstanceAutomationLabelsOnTermination(instance.GetLabels()))
+	submissionRequired := false
+
+	template, err := RetrieveTemplate(ctx, r.Client, instance)
+
+	if err != nil {
+		log.Error(err, "failed retrieving template")
+		_, err := CheckEnvironmentValidity(instance, template)
+		if err == nil {
+			submissionRequired = true
+			log.Info("submission required")
+		} else {
+			log.Info("environment is not valid for submission", "error", err)
```

Comment:
I suppose you should return the error

Reply:
This occurs also for instances that do not have the url set, is it still good to put it as an error?

Reply:
I mean only in case the error is related to the failure in retrieving the environment.